tests/test_sparse_gp.cc

```diff
@@ -204,4 +204,108 @@ TYPED_TEST(SparseGaussianProcessTest, test_likelihood) {
   EXPECT_NEAR(expected, actual, 1e-2);
 }
 
+struct FixedInducingPoints {
+
+  FixedInducingPoints(double min, double max, std::size_t num_points = 10)
+      : min_(min), max_(max), num_points_(num_points) {}
+
+  template <typename CovarianceFunction>
+  std::vector<double> operator()(const CovarianceFunction &cov,
+                                 const std::vector<double> &features) const {
+    return linspace(min_, max_, num_points_);
+  }
+
+  double min_;
+  double max_;
+  std::size_t num_points_;
+};
+
+TEST(test_sparse_gp, test_update_exists) {
+
+  using CovFunc = SquaredExponential<EuclideanDistance>;
+  using MeanFunc = ZeroMean;
+  using GrouperFunction = LeaveOneIntervalOut;
+  using InducingPointStrategy = UniformlySpacedInducingPoints;
+
+  using SparseGPR =
+      SparseGaussianProcessRegression<CovFunc, MeanFunc, GrouperFunction,
+                                      InducingPointStrategy>;
+
+  using FitType = Fit<SparseGPFit<double>>;
+
+  EXPECT_TRUE(bool(has_valid_update<SparseGPR, FitType, double>::value));
+  EXPECT_TRUE(bool(can_update_in_place<SparseGPR, FitType, double>::value));
+}
+
+TYPED_TEST(SparseGaussianProcessTest, test_update) {
+  auto grouper = this->grouper;
+  auto covariance = make_simple_covariance_function();
+  auto dataset = make_toy_linear_data();
+
+  const double min =
+      *std::min_element(dataset.features.begin(), dataset.features.end());
+  const double max =
+      *std::max_element(dataset.features.begin(), dataset.features.end());
+
+  FixedInducingPoints strategy(min, max, 8);
+  auto sparse =
+      sparse_gp_from_covariance(covariance, grouper, strategy, "sparse");
+  sparse.set_param(details::inducing_nugget_name(), 1e-3);
+  sparse.set_param(details::measurement_nugget_name(), 1e-12);
+
+  auto groups = dataset.group_by(grouper).groups();
+  const std::size_t original_size = groups.size();
+
+  const auto held_out_pair = groups.first_group();
+  groups.erase(held_out_pair.first);
+
+  ASSERT_LT(groups.size(), original_size);
+
+  const auto partial_dataset = groups.combine();
+
+  const auto full_dataset =
+      albatross::concatenate_datasets(partial_dataset, held_out_pair.second);
+
+  auto full_fit = sparse.fit(full_dataset);
+  auto partial_fit = sparse.fit(partial_dataset);
+
+  // Copy the fit so we can update it in place
+  decltype(partial_fit) updated_fit(partial_fit);
```

Comment:
What happens when we try `partial_fit.update_in_place()`? `partial_fit` isn't const.

Reply:
That'd work fine (and would be the actual work flow) here I just wanted to leave it partially fit so we can make sure it's different than the full prediction.